doc/man3/flux_future_create.adoc

```diff
@@ -0,0 +1,117 @@
+flux_future_create(3)
+=====================
+:doctype: manpage
+
+
+NAME
+----
+flux_future_create, flux_future_destroy, flux_future_get, flux_future_check, flux_future_then, flux_future_set_timeout - synchronize an activity
+
+
+SYNOPSIS
+--------
+ #include <flux/core.h>
+
+ typedef struct flux_future flux_future_t;
+
+ typedef void (*flux_continuation_f)(flux_future_t *f, void *arg);
+
+ typedef void (*flux_future_init_f)(flux_future_t *f,
+                                    flux_reactor_t *r, void *arg);
+
+ flux_future_t *flux_future_create (flux_future_init_f *cb, void *arg);
```

Comment:
I guess I'm still confused by the complexity here.  Is there a reason that we can't just pass a callback and a reactor right into the flux_future_create(), and then completely eliminate all the flux_future_then() business?  The flux_future_init_f() will be doing the flux_future_set_result(), right (or I guess really the watcher that it sets up will do the flux_future_set_result()).  So why set up potentially a second callback to do what the first callback could have done with less complexity?